CMSIS/NN/Tests/UnitTest/TestCases/test_arm_convolve_s8/test_arm_convolve_s8.c

```diff
@@ -104,3 +108,225 @@ void stride2pad1_arm_convolve_s8(void)
+                                      CONV_3_OUT_ACTIVATION_MAX,
+                                      CONV_3_OUTPUT_W,
+                                      CONV_3_OUTPUT_H,
+                                      bufferA);
+
+  TEST_ASSERT_EQUAL(expected, result);
+  TEST_ASSERT_TRUE(validate(output, conv_3_output_ref, CONV_3_DST_SIZE));
+
+  free(bufferA);
+}
+
+void conv_1_x_n_1_arm_convolve_s8(void)
+{
+  arm_status expected = ARM_MATH_SUCCESS;
+  q7_t output[CONV_1_X_N_1_DST_SIZE] = {0};
+
+  const int32_t buf_size = arm_convolve_s8_get_buffer_size(CONV_1_X_N_1_IN_CH,
+                                                           CONV_1_X_N_1_FILTER_X,
+                                                           CONV_1_X_N_1_FILTER_Y);
+  q15_t *bufferA = (q15_t*)malloc(buf_size);
+
+  arm_status result = arm_convolve_s8(conv_1_x_n_1_input,
+                                      CONV_1_X_N_1_INPUT_W,
+                                      CONV_1_X_N_1_INPUT_H,
+                                      CONV_1_X_N_1_IN_CH,
+                                      CONV_1_X_N_1_INPUT_BATCHES,
+                                      conv_1_x_n_1_weights,
+                                      CONV_1_X_N_1_OUT_CH,
+                                      CONV_1_X_N_1_FILTER_X,
+                                      CONV_1_X_N_1_FILTER_Y,
+                                      CONV_1_X_N_1_PAD_X,
+                                      CONV_1_X_N_1_PAD_Y,
+                                      CONV_1_X_N_1_STRIDE_X,
+                                      CONV_1_X_N_1_STRIDE_Y,
+                                      conv_1_x_n_1_biases,
+                                      output,
+                                      conv_1_x_n_1_output_shift,
+                                      conv_1_x_n_1_output_mult,
+                                      CONV_1_X_N_1_OUTPUT_OFFSET,
+                                      CONV_1_X_N_1_INPUT_OFFSET,
+                                      CONV_1_X_N_1_OUT_ACTIVATION_MIN,
+                                      CONV_1_X_N_1_OUT_ACTIVATION_MAX,
+                                      CONV_1_X_N_1_OUTPUT_W,
+                                      CONV_1_X_N_1_OUTPUT_H,
+                                      bufferA);
+
+  TEST_ASSERT_EQUAL(expected, result);
+  TEST_ASSERT_TRUE(validate(output, conv_1_x_n_1_output_ref, CONV_1_X_N_1_DST_SIZE));
+
+  free(bufferA);
+}
+
+void conv_1_x_n_1_1_arm_convolve_s8(void)
+{
+  arm_status expected = ARM_MATH_SIZE_MISMATCH;
+  q7_t output[CONV_1_X_N_1_DST_SIZE] = {0};
+
+  const int32_t buf_size = arm_convolve_1_x_n_s8_get_buffer_size(CONV_1_X_N_1_IN_CH,
+                                                                 CONV_1_X_N_1_FILTER_X,
+                                                                 CONV_1_X_N_1_FILTER_Y);
+  q15_t *bufferA = (q15_t*)malloc(buf_size);
+
+  arm_status result = arm_convolve_1_x_n_s8(conv_1_x_n_1_input,
```

Comment:
arm_convolve_1_x_n_s8 should perhaps be in its own file?
TestCases/test_arm_convolve_1_x_n_s8/test_arm_convolve_1_x_n_s8.c

Reply:
Atleast for the DSP extension(which is the target here), it uses arm_convolve_s8 at the backend. So,  went with the same file..

Reply:
Ok, let's go for that